src/main/java/redis/clients/jedis/JedisClusterCommand.java

```diff
@@ -119,22 +116,25 @@ private T runWithRetries(final int slot) {
           connection = connectionHandler.getConnectionFromSlot(slot);
         }
 
-        return execute(connection);
+        try {
+          return execute(connection);
+        } catch (JedisConnectionException e) {
+          LOG.debug("Connecting to {} failed once, shortcut retrying without backoff...", connection, e);
+          T result = execute(connection);
+          int doneAttempts = 1 + maxAttempts - attemptsLeft;
+          LOG.debug("Shortcut retry worked, returning result after {} attempts", doneAttempts);
+          return result;
```

Comment:
1. It can be done only if attempts are left.
2. It should reflect in attempts left calculation.

Reply:
Then I think the other approach is cleaner, so I will close this PR!